core/src/test/java/com/linecorp/armeria/client/retry/RetyingClientEventLoopSchedulerTest.java

```diff
@@ -0,0 +1,108 @@
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.http(0);
+            sb.http(0);
+            sb.http(0);
+            sb.service("/fail", (ctx, req) -> {
+                throw new AnticipatedException();
+            });
+            sb.service("/ok", (ctx, req) -> {
+                return HttpResponse.of(200);
+            });
+        }
+    };
+
+    @Test
+    void shouldReturnCorrectEventLoop() {
+        final List<Endpoint> endpoints = server.server().activePorts().values().stream()
+                                               .map(port -> Endpoint.of(port.localAddress()))
+                                               .collect(toImmutableList());
+        assertThat(endpoints).hasSize(3);
+        final Map<Endpoint, EventLoop> eventLoopMapping = new HashMap<>();
+
+        for (Endpoint endpoint : endpoints) {
+            // Acquire the event loops for each endpoint.
+            try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+                final AggregatedHttpResponse res = WebClient.of(SessionProtocol.H2C, endpoint)
+                                                            .blocking()
+                                                            .get("/ok");
+                assertThat(res.status()).isEqualTo(HttpStatus.OK);
+                eventLoopMapping.put(endpoint, captor.get().eventLoop().withoutContext());
+            }
+        }
+
+        // Check that the event loops are correctly mapped for each attempt.
+        final EndpointGroup endpointGroup = EndpointGroup.of(endpoints);
+        final RetryRule retryRule = RetryRule.builder()
+                                             .onServerErrorStatus()
+                                             .thenBackoff(Backoff.withoutDelay());
+        final BlockingWebClient client =
+                WebClient.builder(SessionProtocol.H2C, endpointGroup)
+                         // Make retries until the maxTotalAttempts is reached.
+                         .responseTimeoutMillis(0)
+                         .decorator(RetryingClient.newDecorator(
+                                 RetryConfig.builder(retryRule)
+                                            .maxTotalAttempts(6)
+                                            .build()))
+                         .build()
+                         .blocking();
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            assertThat(client.get("/fail").status()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
+            assertThat(captor.get().log().children().size()).isEqualTo(6);
+            for (int i = 0; i < 6; i++) {
+                final RequestContext childCtx = captor.get().log().children().get(i).context();
+                assertThat(childCtx.eventLoop().withoutContext())
+                        .isEqualTo(eventLoopMapping.get(endpoints.get(i % 3)));
```

Comment:
Should we add one more assert and use `isSameAs()` just in case?
```suggestion
                final Endpoint expectedEndpoint = eventLoopMapping.get(endpoints.get(i % 3));
                assertThat(childCtx.endpoint()).isSameAs(expectedEndpoint);
                assertThat(childCtx.eventLoop().withoutContext())
                        .isSameAs(eventLoopMapping.get(expectedEndpoint));
```